### PR TITLE
map JSON data must be used raw -- do not escape it

### PR DESCRIPTION
**Pull request by weedySeaDragon**
Bugfix:  companies map was not displaying correctly with ruby 2.4.0 and rails 5.0.1

PT Story: https://www.pivotaltracker.com/story/show/147250491

Rails now escapes strings by default.   The JSON data for the company markers on the map were being escaped, which 'broke' the map.

Changes proposed in this pull request:
1.  use 'raw' to be sure that the JSON data is _not_ automatically escaped

Ready for review:
@thesuss @RobertCram @patmbolger

### CODE REVIEW

**Comment by patmbolger** [comment]
How did this occur with the upgrade to Ruby 2.4?  We did not change the Rails version ....

**Comment by weedySeaDragon** [comment]
@patmbolger Hm. Obv I didn't think thru this deeply enough.
Looks like my version of the `json` gem is later (>) than the previous one in the Gemfile.lock 

I'll investigate to see if the `json` gem version is the culprit.  (there is the Ruby language `json` stuff, the Rails `json` stuff, and the `json` gem stuff.  <sigh>)

**Comment by weedySeaDragon** [comment]
I've spent a few hours on this and I can't tell why it started failing for me (or why it was actually working).

Should really be using `raw .....` to ensure that the json data is not escaped anyway.

And I've now included an RSPec test to check that it always *is* escaped, so if we run into some weirdness again, we'll know.

File: spec/views/companies/_map_companies.haml_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+class TempCompanyHelper
+  include CompaniesHelper
+end
+
+RSpec.describe "company maps partial: _map_companies.html.haml" do
+
+  it "map markers are raw json (not escaped)" do
+
+    test_companies = []
+    test_companies << create(:company)
```
Comment (patmbolger) [inline spec/views/companies/_map_companies.haml_spec.rb]: You could collapse these two statements into:
```
test_companies = [ create(:company) ]
```
but that is simply a matter of preference.

Comment (patmbolger) [reply spec/views/companies/_map_companies.haml_spec.rb]: Also, to be consistent with naming convention, this file should be named:
```
_map_companies.html.haml_spec.rb
```
It still works regardless, but its nice to be consistent :)

**Review by patmbolger — COMMENTED** [review]
Please see comments.

**Comment by patmbolger** [comment]
This is a real puzzle.  I pulled this PR and could not duplicate any problem with map display.  

Also, I can't see how Rails would even escape the map HTML, as the map is actually generated and rendered by the Google API (?)

**Comment by weedySeaDragon** [comment]
It's not the actual map -- it's the data that is _passed_ to the map:  the information about the markers ("pins") to be displayed on the map (e.g. the location coordinates, what to display in the pop-up box when the user clicks on a marker, etc.)
That data is returned by the `CompaniesController` and then converted to JSON with the `.to_json` method.  It's that data -- the marker data as JSON -- that was being escaped.

I think it must have been some particular combination/ versions of the gems that I have on my system.

At any rate, we _should_ be using`raw` in the javascript.

File: spec/views/companies/_map_companies.haml_spec.rb (hunk 1 above)
Comment (weedySeaDragon) [reply spec/views/companies/_map_companies.haml_spec.rb]: Agreed.  It should definitely be renamed.

**Comment by patmbolger** [comment]
OK, now I understand.  BTW, did you mean to add the `raw` method to the JS?  I don't see that in this PR.

**Comment by weedySeaDragon** [comment]
Yup -it's in there. Probably you're just not seeing it in the line for the latest commit I did.  It was changed in the first commit:
https://github.com/weedySeaDragon/shf-project/blob/cd97e6769000780c8032ebcf721b3f85e3251ec6/app/views/companies/_map_companies.haml#L8

**Review by RobertCram — APPROVED** [review]

**Review by patmbolger — APPROVED** [review]

**Pull request merged**